readLog/src/main/java/org/corfudb/readLog/readLog.java

```diff
@@ -0,0 +1,354 @@
+        ByteBuffer comma_buffer = ByteBuffer.allocate(2);
+        int bytes_read = fileChannelIn.read(comma_buffer);
+        comma_buffer.flip();
+        Short delim = comma_buffer.getShort();
+        comma_buffer.flip();
+        if (delim != DELIMITER) {
+            System.out.println("Incorrect delimiter");
+        }
+        ByteBuffer md_buffer = ByteBuffer.allocate(MetadataSize);
+        bytes_read += fileChannelIn.read(md_buffer);
+        md_buffer.flip();
+        Metadata md = Metadata.parseFrom(md_buffer.array());
+        ByteBuffer record_buffer = ByteBuffer.allocate(md.getLength());
+        bytes_read += fileChannelIn.read(record_buffer);
+        record_buffer.flip();
+        int cksum = getChecksum(record_buffer.array());
+        if (cksum != md.getChecksum()) {
+            System.out.println("Checksum ERROR");
+        }
+        LogEntry le_new = processRecordBody(md, record_buffer);
+        return new LogEntryExtended(le_new, bytes_read, cksum);
+    }
+
+    void openLogFile(boolean display) throws IOException {
+        logHeader hdr = processHeader();
+        if (display) {
+            System.out.println("checksum " + String.format("%08x", hdr.getChecksum()));
+            System.out.println("length " + Integer.toString(hdr.getLength()));
+            System.out.println("version " + Integer.toString(hdr.getVersion()));
+            System.out.println("verify " + Boolean.toString(hdr.isVerifyChecksum()));
+        }
+        remSize = fileChannelIn.size() - fileChannelIn.position();  // if size == position then file pointer is off the end
+    }
+
+    LogEntryExtended nextRecord(boolean display) throws IOException {
+        LogEntryExtended le_new = processRecord();
+        if (display) {
+            System.out.format("Record length %d checksum %08x\n", le_new.getBytesLength(), le_new.getChecksum());
+        }
+        if (display) {
+            printLogEntry(le_new.getEntryBody());
+        }
+        if (fileChannelOut != null) {
+            writeBuffer(le_new.getEntryBody(), fileChannelOut);
+        }
+        recordCnt++;
+        remSize -= le_new.getBytesLength();
+        return le_new;
+    }
+
+    int processLogFile() throws IOException {
+        boolean display = op.getOpType() == Operation.OperationType.DISPLAY ||
+                op.getOpType() == Operation.OperationType.DISPLAY_ALL;
+        boolean display_data = op.getOpType() == Operation.OperationType.DISPLAY_ALL;
+
+        System.out.format("processLogFile display %d output %d\n", display ? 1 : 0,
+                fileChannelOut != null ? 1 : 0);
+
+        // Metadata
+        //   LogHeader
+        // Metadata
+        //   LogEntry
+        // ...
+        openLogFile(display);
+        while (remSize > 0) {
+            nextRecord(display);
+        }
+        // REPORT
+        System.out.println("Read " + Integer.toString(recordCnt) + " log entries");
+        return recordCnt;
+    }
+
+    Operation op;
```

Comment:
Move class variables before the constructor. Also,  non primitive types are initialized to null.